File: src/merge_strings_alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        ans = ""
        i1, i2 = 0, 0
        for i in range(len(word1) + len(word2)):
            if i%2 == 0:
                ans += word1[i1]
                i1 += 1
            else:
                ans += word2[i2]
                i2 += 1
            
            if i1 >= len(word1):
                ans += word2[i2:]
                break
            elif i2 >= len(word2):
                ans += word1[i1:]
                break
        
        return ans

s = Solution()
print(s.mergeAlternately("abc", "pqr"))
print(s.mergeAlternately("ab", "pqrs"))
print(s.mergeAlternately("abcd", "pq"))

File: java/MergeStringsAlternately.java
public class MergeStringsAlternately {
  public static void main(String[] args) {
    System.out.println(run("abc", "pqr"));
    System.out.println(run("ab", "pqrs"));
    System.out.println(run("abcd", "pq"));
  }

  static String run(String word1, String word2) {
    var n1 = word1.length();
    var n2 = word2.length();
    var n = Math.max(n1, n2);
    var i = 0;
    var ans = "";
    while (i < n) {
      if (i < n1) {
        ans += word1.charAt(i);
      }
      
      if (i < n2) {
        ans += word2.charAt(i);
      }
      i++;
    }
    return ans;
    }
}
